src/Tests/Microsoft.DotNet.PackageValidation.Tests/PackageValidationTests.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.PackageValidation.Tests
+{
+    public class PackageValidationTests : SdkTest
+    {
+        public TestLogger _logger;
+
+        public PackageValidationTests(ITestOutputHelper log) : base(log) 
+        {
+            _logger = new();
+        }
+
+        [Fact]
+        public void MissingRidLessAssetForFramework()
+        {
+            string[] filePaths = new []
+            {
+                @"ref/netcoreapp3.1/TestPackage.dll", 
+                @"runtimes/win/lib/netcoreapp3.1/TestPackage.dll"
+            };
+
+            Package package = new("TestPackage", "1.0.0", filePaths, null, null);
+            new CompatibleTfmValidator(string.Empty, null, false, _logger).Validate(package);
+            Assert.Single(_logger.errors);
+            Assert.Equal("PKV004 There is no compatible runtime asset for target framework .NETCoreApp,Version=v3.1 in the package.", _logger.errors[0]);
+        }
+
+        [Fact]
+        public void MissingAssetForFramework()
+        {
+            string[] filePaths = new[]
+            {
+                @"ref/netstandard2.0/TestPackage.dll",
+                @"lib/netcoreapp3.1/TestPackage.dll"
+            };
+
+            Package package = new("TestPackage", "1.0.0", filePaths, null, null);
+            new CompatibleTfmValidator(string.Empty, null, false, _logger).Validate(package);
+            Assert.NotEmpty(_logger.errors);
```

Comment:
Are you expecting more, or should this be a single error?

Reply:
There could be more than one errors. This will report errors for all the compatible frameworks.